anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/mapper/CacheUtil.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.mapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * CacheUtil utilities.
+ */
+@Slf4j
+public final class CacheUtil {
+
+    private static final String DELIMITER = "=";
```

Comment:
we need to make sure that tag name or tag value for a metric never has this DELIMITER in it. if not `getTags()` wont work as expected. So it is better to put some validation around this - may be log a warning and ignore if a metric tag has DELIMITER in it?